[docs] Improve Secure saved objects and kibana-encryption-keys docs (#132828)
elastic/kibana: docs/user/security/secure-saved-objects.asciidoc

If you don't specify an encryption key, {kib} might disable features that rely on encrypted saved objects.
============================================================================

[TIP]
============================================================================

> For help generating the encryption key, see the <<kibana-encryption-keys, `kibana-encryption-keys`>> script.

Reviewer: ```suggestion
For help generating the encryption key, refer to the <<kibana-encryption-keys, `kibana-encryption-keys`>> script.
```

============================================================================

At some point, you might want to dispose of old encryption keys completely. Make sure there are no saved objects that {kib} encrypted with these encryption keys. You can use the <<saved-objects-api-rotate-encryption-key, rotate encryption key API>> to determine which existing saved objects require decryption-only keys and re-encrypt them with the primary key.

[[encryption-key-docker-configuration]]
==== Docker configuration

> It's also possible to configure the encryption keys using <<environment-variable-config,Docker Environment Variables>>.

Reviewer: ```suggestion
It's also possible to configure the encryption keys using <<environment-variable-config,Docker environment variables>>.
```

> Docker Environment Variable Examples:

Reviewer: ```suggestion
Docker environment variable examples:
```
